src/player.js

```diff
@@ -1006,7 +1036,13 @@ export default class Player extends FakeEventTarget {
         if (track.language === OFF) {
           this.hideTextTrack();
           this._playbackAttributesState.textLanguage = OFF;
+        } else if (track.external && !this._config.playback.useNativeTextTrack) {
+          this._engine.hideTextTrack();
+          this._externalCaptionsHandler.selectTextTrack(track);
         } else {
+          if (this._externalCaptionsHandler) {
+            this._externalCaptionsHandler.hideTextTrack();
```

Comment:
always call this, and handle if selected track is available in the external captions handler